### PR TITLE
Ng 26 ascii display

### PR DESCRIPTION
**Pull request by noelvalentin**
Added ASCII grid. Added the actual grid rectangles in Builder String and created the ASCIIGrid class for it. Then, created 8 Label widgets in the RunWindow class to show the ASCII interpretation of the content in memories 4088-4095 in RAM. Then, created an update_grid method that is executed whenever the run or debug button have been pressed. This method updates the text of each respective label, and gives it the most updated value. 

The content in memory could have a value too high for an ASCII character, and some ASCII items (such as chr(20),which is DC4) are not visible using the chr() function; apart from that, the feature seems to work appropriately.

### CODE REVIEW

File: GUI/window.py [hunk 1]
```diff
@@ -211,21 +364,35 @@ def run_micro_instructions_step(self, instance):
...
+                                             self.micro_sim.disassembled_instruction())
+                    self.light.change_color(self.micro_sim.traffic_lights_binary())
+                    self.update_grid()
 
                 toast('Runnin instruction in step-by-step mode. Step ' + str(self.step_index) + ' is running')
                 for i in self.micro_sim.micro_instructions:
                     if i != 'NOP':
                         print(i)
 
+    def update_grid(self):
```
Comment (ChristianPerez34) [inline GUI/window.py]: Consider renaming this method. `RunWindow` has various widgets so method should be clear on what widget or in this case I/O it is trying to update the grid for. Another option would be to move this method the `ASCIIGrid`

**Review by ChristianPerez34 — CHANGES_REQUESTED** [review]

**Review by ChristianPerez34 — APPROVED** [review]

File: GUI/window.py [hunk 2]
```diff
@@ -204,14 +243,33 @@ def __init__(self, **kwargs):
...
+                                   font_size=40, markup=True)
+        self.ascii_label_5 = Label(text='[color=000000]' + chr(int(RAM[4092], 16)) + '[/color]', pos=(-8, 29),
+                                   font_size=40, markup=True)
+        self.ascii_label_6 = Label(text='[color=000000]' + chr(int(RAM[4093], 16)) + '[/color]', pos=(38, 29),
+                                   font_size=40, markup=True)
+        self.ascii_label_7 = Label(text='[color=000000]' + chr(int(RAM[4094], 16)) + '[/color]', pos=(84, 29),
+                                   font_size=40, markup=True)
+        self.ascii_label_8 = Label(text='[color=000000]' + chr(int(RAM[4095], 16)) + '[/color]', pos=(130, 29),
+                                   font_size=40, markup=True)
+
```
Comment (alejoreyes96) [inline GUI/window.py]: The positions need to be remodeled due to merging of the hex keyboard. Id say to place them below the keyboard; merge with master and update

**Review by alejoreyes96 — CHANGES_REQUESTED** [review]

File: GUI/window.py [hunk 3]
```diff
@@ -598,6 +679,11 @@ def change_color(self, binary):
                     self.green_1 = (0,1,0)
         
 
+class ASCIIGrid(Widget):
+
+    def __init__(self, **kwargs):
+        super(ASCIIGrid, self).__init__(**kwargs)
+
```
Comment (alejoreyes96) [inline GUI/window.py:993]: This class is inherently doing nothing, just passing the widget. I think you could move all the labels declared on the run window to this method, then add them to the widget using `self.add_widget`. Then, you could pass the main widget to the run window (like you are already doing) and then add that main widget to the run window. You would also need to change the update method down to that class, and change the calling of the method from `self.update_ascii_grid()` to `self.ascii.update_ascii_grid()`.

Something like this, but with all the labels:

`class ASCIIGrid(Widget):

    def __init__(self, **kwargs):
        super(ASCIIGrid, self).__init__(**kwargs)
        self.ascii_label_1 = Label(text='[color=000000]' + chr(int(RAM[4088], 16)) + '[/color]', pos=(-187, 29),
                                   font_size=40, markup=True)
        self.add_widget(self.ascii_label_1)
    
    def update_ascii_grid(self):
        self.ascii_label_1.text = '[color=000000]' + chr(int(RAM[4088], 16)) + '[/color]'`

**Review by alejoreyes96 — CHANGES_REQUESTED** [review]

File: GUI/window.py (hunk 2 above)
Comment (noelvalentin) [reply GUI/window.py]: Thank you for the suggestion, the ASCII grid's position has been updated to be under the Hex Keyboard in the latest commit

File: GUI/window.py (hunk 3 above)
Comment (alejoreyes96) [reply GUI/window.py:993]: Was verified and deduced it is alright to leave as is for now.

**Review by alejoreyes96 — APPROVED** [review]

**Pull request merged**